### PR TITLE
fix: check if url local with platform custom protocol, closes #7176

### LINKED ISSUE #7176: [bug] invoke broken on android dev
### Describe the bug

As of this commit a50f24b2bd93864cacd9047b6fcd152f71a38c45 calling invoke from the web browser no longer works while running the application in an android emulator using `cargo tauri android dev`.

My target is android, I'm running on MacOS and my front end is built using Sycamore.

I used `git bisect` to determine when the next branch first broke for me and determined that it was the merge commit that I mentioned. There's not an issue with the desktop application in dev mode, just the android one. I haven't tracked down where the issue was coming from, but figured I would write up the issue now in case someone has a quick idea on what the issue could be and where to look before I have a chance to finish running down this bug.

### Reproduction

This can be easily reproduced with the sample app with a mobile dev environment already setup.

```bash
sh <(curl https://create.tauri.app/sh) --alpha --mobile --manager cargo --template sycamore test
cd test
cargo tauri android init
tee -a Cargo.toml <<EOF
[patch.crates-io]
tauri = { git = "https://github.com/tauri-apps/tauri.git", branch = "next" }
EOF
cargo tauri android dev
```

### Expected behavior

_No response_

### Platform and versions

```text
[✔] Environment
    - OS: Mac OS 12.6.3 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.69.0 (84c898d65 2023-04-16)
    ✔ Cargo: 1.69.0 (6e9a83356 2023-04-12)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
    - node: 19.9.0
    - pnpm: 7.27.0
    - yarn: 1.22.19
    - npm: 9.6.3

[-] Packages
    - tauri [RUST]: 2.0.0-alpha.9
    - tauri-build [RUST]: no manifest (2.0.0-alpha.5, 2.0.0-alpha.5)
    - wry [RUST]: 0.28.3
    - tao [RUST]: 0.19.1
    - @tauri-apps/api [NPM]: not installed!
    - @tauri-apps/cli [NPM]: 2.0.0-alpha.9

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: core/tauri/src/window.rs [hunk 1]
```diff
@@ -1653,13 +1653,18 @@ impl<R: Runtime> Window<R> {
     self.current_url = url;
   }
 
+  fn is_local_url(&self, current_url: &Url) -> bool {
+    self.manager.get_url().make_relative(&current_url).is_some() || {
```
Comment (amrbashir) [inline core/tauri/src/window.rs]:
```suggestion
    self.manager.get_url().make_relative(current_url).is_some() || {
```

**Comment by lucasfernog** [comment]
I believe I pushed some fixes on the custom protocol IPC PR too.

**Review by lucasfernog — APPROVED** [review]
But this is much better :)

**Comment by amrbashir** [comment]
> I believe I pushed some fixes on the custom protocol IPC PR too.

yeah I was originally gonna link the issue to your PR but it wasn't enough to fix the issue so I had to make another PR

**Pull request merged**